drivers/hd44780/hd44780.c

```diff
@@ -110,19 +110,17 @@ int hd44780_init(hd44780_t *dev, const hd44780_params_t *params)
         LOG_ERROR("hd44780_init: invalid LCD size!\n");
         return -1;
     }
-    uint8_t count_pins = 0;
-    /* check which pins are used */
+    /* Copy configured pins in device descriptor */
     for (unsigned i = 0; i < HD44780_MAX_PINS; ++i) {
-        if (dev->p.data[i] != GPIO_UNDEF) {
-            ++count_pins;
-        }
+        dev->data[i] = dev->p.data[i];
```

Comment:
any reason no `memcpy(dev->data, dev.p.data, ARRAY_SIZE((dev->data));`  (I don't know if that is the proper array size macro but you get the idea)?

Reply:
I tried memcpy and noticed a bigger code size on ROM, that's why I kept the for loop version.

Reply:
Fair enough, lets me run it through the test.